core/store/migrations/migration1597695690/migrate.go

```diff
@@ -0,0 +1,194 @@
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN is_bootstrap_peer bool NOT NULL;
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN encrypted_ocr_key_bundle_id bytea NOT NULL REFERENCES encrypted_ocr_key_bundles (id);
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN monitoring_endpoint TEXT;
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN transmitter_address bytea;  -- NOT NULL REFERENCES keys (address)
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN observation_timeout bigint NOT NULL;
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN blockchain_timeout bigint NOT NULL;
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN contract_config_tracker_subscribe_interval bigint NOT NULL;
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN contract_config_tracker_poll_interval bigint NOT NULL;
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN contract_config_confirmations INT NOT NULL;
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN created_at timestamptz NOT NULL;
+        ALTER TABLE offchainreporting_oracle_specs ADD COLUMN updated_at timestamptz NOT NULL;
+        ALTER TABLE offchainreporting_oracle_specs ADD CONSTRAINT chk_contract_address_length CHECK (octet_length(contract_address) = 20);
+
+        -- NOTE: This will be extended with new IDs when we bring directrequest and fluxmonitor under the new jobspawner umbrella
+        -- Only ONE id should ever be present
+        CREATE TABLE jobs (
+            id SERIAL PRIMARY KEY,
+            pipeline_spec_id INT REFERENCES pipeline_specs (id),
+            offchainreporting_oracle_spec_id INT REFERENCES offchainreporting_oracle_specs (id),
+            CONSTRAINT chk_valid CHECK (
+                offchainreporting_oracle_spec_id IS NOT NULL
+            )
+        );
+
+        ---
+        --- Notify the Chainlink node when a new job spec is created
+        ---
+
+        CREATE OR REPLACE FUNCTION notifyJobCreated() RETURNS TRIGGER AS $_$
+        BEGIN
+            PERFORM pg_notify('job_created', NEW.id::text);
+            RETURN NEW;
+        END
+        $_$ LANGUAGE 'plpgsql';
+
+        CREATE TRIGGER notify_job_created
+        AFTER INSERT ON jobs
+        FOR EACH ROW EXECUTE PROCEDURE notifyJobCreated();
+
+
+        ---
+        --- Delete job specs from their respective tables when a row from 'jobs' is deleted
+        ---
+
+        CREATE OR REPLACE FUNCTION deleteConcreteSpecs() RETURNS TRIGGER AS $_$
+        BEGIN
+            DELETE FROM offchainreporting_oracle_specs WHERE offchainreporting_oracle_specs.id = OLD.offchainreporting_oracle_spec_id;
+            DELETE FROM pipeline_specs WHERE pipeline_specs.id = OLD.pipeline_spec_id;
+            RETURN OLD;
+        END $_$ LANGUAGE 'plpgsql';
+
+        CREATE TRIGGER delete_job_components
+        AFTER DELETE ON jobs
+        FOR EACH ROW EXECUTE PROCEDURE deleteConcreteSpecs();
```

Comment:
This is really smelly. It probably means you're deleting the wrong record on the software side.

If you delete the spec the job will automatically be removed due to foreign key cascade.